hypnos/src/event_handler.c

```diff
@@ -33,15 +40,27 @@ void event_handler_init()
 	/* Initialize GPIOs */
         charging_dev = device_get_binding("GPIO_0");
         gpio_pin_configure(charging_dev, BAT_CHA_PIN, GPIO_DIR_IN | GPIO_INT
-                | GPIO_INT_EDGE | GPIO_INT_DOUBLE_EDGE);
+                | EDGE);
         gpio_init_callback(&charging_cb, battery_callback_charging,
                 BIT(BAT_CHA_PIN));
+	button_dev = device_get_binding(BUT_PORT);
+	gpio_pin_configure(button_dev, BUT_PIN, GPIO_DIR_IN | GPIO_INT |  PULL_UP
+			   | EDGE);
+	gpio_init_callback(&button_cb, button_callback, BIT(BUT_PIN));
 
 	/* Enable GPIOs */
 	gpio_add_callback(charging_dev, &charging_cb);
         gpio_pin_enable_callback(charging_dev, BAT_CHA_PIN);
+	gpio_add_callback(button_dev, &button_cb);
+        gpio_pin_enable_callback(button_dev, BUT_PIN);
+
+	/* Set port 15 to high to enable the button */
+	u32_t button_out = 1U;
+        gpio_pin_configure(button_dev, 15, GPIO_DIR_OUT); //push button out
```

Comment:
Use alias for pin 15

Reply:
I fixed it by defining BTN_IN and BTN_OUT. Then I also removed the PIN suffix which I felt is redundant. Are you OK with that?